pkg/node/kubelet.go

```diff
@@ -47,17 +86,50 @@ func StartKubelet(cfg *config.MicroshiftConfig) error {
 		args = append(args, "--log-file="+filepath.Join(cfg.LogDir, "kubelet.log"))
 	}
 
-	if err := command.ParseFlags(args); err != nil {
-		logrus.Fatalf("failed to parse flags:%v", err)
+	cmd := &cobra.Command{
+		Use:          "kubelet",
+		Long:         `kubelet`,
+		SilenceUsage: true,
+		RunE:         func(cmd *cobra.Command, args []string) error { return nil },
 	}
-	logrus.Infof("starting kubelet %s, args: %v", cfg.HostIP, args)
 
-	go func() {
-		command.Run(command, args)
-		logrus.Fatalf("kubelet exited")
-	}()
+	cleanFlagSet := pflag.NewFlagSet(componentKubelet, pflag.ContinueOnError)
+	cleanFlagSet.SetNormalizeFunc(cliflag.WordSepNormalizeFunc)
+	kubeletFlags := kubeletoptions.NewKubeletFlags()
+	kubeletFlags.AddFlags(cleanFlagSet)
+	kubeletoptions.AddKubeletConfigFlags(cleanFlagSet, kubeletConfig)
+	kubeletoptions.AddGlobalFlags(cleanFlagSet)
 
+	if err := cmd.ParseFlags(args); err != nil {
+		logrus.Fatalf("failed to parse flags:%v", err)
+		return err
+	}
+	s.kubeletflags = kubeletFlags
+	s.kubeconfig = kubeletConfig
+	s.kubeconfigfile = filepath.Join(cfg.DataDir, "resources", "kubelet", "kubeconfig")
+	logrus.Infof("Starting kubelet %s, args: %v", cfg.HostIP, args)
```

Comment:
HostIP -> NodeIP

Reply:
It created a merge conflict. I have rebased the PR branch